compiler_gym/envs/compiler_env.py

```diff
@@ -3,85 +3,21 @@
 # LICENSE file in the root directory of this source tree.
 """This module defines the OpenAI gym interface for compilers."""
-import logging
-import numbers
-import warnings
-from collections.abc import Iterable as IterableType
-from copy import deepcopy
-from math import isclose
-from pathlib import Path
-from time import time
-from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
+from abc import ABC, abstractmethod
+from typing import Iterable, List, Optional, Union
 
 import gym
-import numpy as np
-from deprecated.sphinx import deprecated
 from gym.spaces import Space
 
 from compiler_gym.compiler_env_state import CompilerEnvState
-from compiler_gym.datasets import Benchmark, Dataset, Datasets
-from compiler_gym.datasets.uri import BenchmarkUri
-from compiler_gym.service import (
-    CompilerGymServiceConnection,
-    ConnectionOpts,
-    ServiceError,
-    ServiceOSError,
-    ServiceTransportError,
-    SessionNotFound,
-)
-from compiler_gym.service.connection import ServiceIsClosed
-from compiler_gym.service.proto import AddBenchmarkRequest
-from compiler_gym.service.proto import Benchmark as BenchmarkProto
-from compiler_gym.service.proto import (
-    EndSessionReply,
-    EndSessionRequest,
-    Event,
-    ForkSessionReply,
-    ForkSessionRequest,
-    GetVersionReply,
-    GetVersionRequest,
-    SendSessionParameterReply,
-    SendSessionParameterRequest,
-    SessionParameter,
-    StartSessionRequest,
-    StepReply,
-    StepRequest,
-    proto_to_action_space,
-)
-from compiler_gym.spaces import DefaultRewardFromObservation, NamedDiscrete, Reward
-from compiler_gym.util.gym_type_hints import (
-    ActionType,
-    ObservationType,
-    RewardType,
-    StepType,
-)
-from compiler_gym.util.shell_format import plural
-from compiler_gym.util.timer import Timer
-from compiler_gym.validation_error import ValidationError
+from compiler_gym.datasets import Benchmark, BenchmarkUri, Dataset
+from compiler_gym.spaces import Reward
+from compiler_gym.util.gym_type_hints import ActionType, ObservationType, StepType
 from compiler_gym.validation_result import ValidationResult
-from compiler_gym.views import ObservationSpaceSpec, ObservationView, RewardView
+from compiler_gym.views import ObservationSpaceSpec
 
-logger = logging.getLogger(__name__)
 
-# NOTE(cummins): This is only required to prevent a name conflict with the now
-# deprecated CompilerEnv.logger attribute. This can be removed once the logger
-# attribute is removed, scheduled for release 0.2.3.
-_logger = logger
-
-
-def _wrapped_step(
-    service: CompilerGymServiceConnection, request: StepRequest
-) -> StepReply:
-    """Call the Step() RPC endpoint."""
-    try:
-        return service(service.stub.Step, request)
-    except FileNotFoundError as e:
-        if str(e).startswith("Session not found"):
-            raise SessionNotFound(str(e))
-        raise
-
-
-class CompilerEnv(gym.Env):
+class CompilerEnv(gym.Env, ABC):
```

Comment:
I think there are still a few properties missing from this interface: `version`, `compiler_version` (possibly others)

Reply:
I added them and a few more.